src/Common/Core/Impl/Collections/DictionaryExtensions.cs

```diff
@@ -8,6 +8,11 @@
 
 namespace Microsoft.Common.Core {
     public static class DictionaryExtensions {
+        public static void Deconstruct<TK, TV>(this KeyValuePair<TK, TV> kvp, out TK key, out TV value) {
```

Comment:
GetParts? `Deconstruct` sounds a bit like it is deleting the pair, but up to you.